consul/src/main/scala/io/buoyant/consul/v1/KvApi.scala

```diff
@@ -95,15 +107,30 @@ class KvApi(
     path: String,
     datacenter: Option[String] = None,
     cas: Option[String] = None,
+    recurse: Boolean = false,
     retry: Boolean = false
   ): Future[Boolean] = {
     val req = mkreq(
       http.Method.Delete,
       s"$kvPrefix$path",
       "cas" -> cas,
+      "recurse" -> (if (recurse) Some(recurse.toString) else None),
       "dc" -> datacenter
     )
-    executeJson[Boolean](req, retry).map(_.value)
+    execute[Boolean](req, retry).map(_.value)
   }
+}
+
+object Key {
+  def apply(key: String, value: String): Key = Key(Some(key), Some(Base64.getEncoder.encodeToString(value.getBytes)))
```

Comment:
It's a little confusing that `Key(k, v)` expects a unencoded value but `Key(Some(k), Some(v))` expects a base64 encoded value.  Consider naming this something other than `apply` (perhaps `mk`?).  This reinforces the idea that a `Key` stores base64 encoded values, but it has a helper method to construct it from an unencoded value.